### ISSUE TITLE
Wayland version looks out of place with the rest of the desktop — missing window shadows

### ISSUE THREAD

**Issue by 1player** [wez/wezterm#1969]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### WezTerm version

wezterm 20220424-164129-3dfe7536

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Tested on Fedora 36 GNOME

When Wayland is enabled, the window doesn't have the same drop shadow as all other applications have, and looks out of place.

### To Reproduce

* Enable Wayland

### Configuration

`enable_wayland=true`

### Expected Behavior

The wezterm window to have the same drop shadow as all other apps.

### Anything else?

wezterm on the left, GNOME Terminal on the right

![image](https://user-images.githubusercontent.com/690120/167602916-2c11e83d-b624-4c3d-8362-8ca9d4687943.png)

**Comment by 1player** [comment]
Don't quote me on that, but AFAIK [libdecor](https://gitlab.gnome.org/jadahl/libdecor) should help with that (render clientside decorations that match the desktop)

**Comment by wez** [comment]
Wayland is weird like this!

If libdecor can improve things, then we're waiting on either one of:
* https://github.com/Smithay/client-toolkit/issues/196
* https://github.com/cmeissl/libdecor-rs

to be ready before we can start to investigate how to integrate this.

Otherwise: it is a non-goal for wezterm to directly try to match the gnome/gtk look.

**Comment by VarLad** [comment]
@wez I wouldn't call not-matching GTK style a bug.
Alacritty and some other applications look different as well
If GNOME really wanted all applications to have the same window decorations, they wouldn't advocate CSD and not implement Server Side Decorations. 
This probably looks standard in the Wayland compositors that support SSD.

**Comment by tmpm697** [comment]
any chance to have `wezterm` decorate its own shadow border? this is headache when i use alacritty on swaywm.

**Comment by wez** [comment]
@tmpm697 I have no plans to do anything about this directly in wezterm. I'm waiting on the issues I mentioned above before coming back to this issue.

**Comment by arashm** [comment]
There seems to be a GTK plugin implemented in libdecor, but there hasn't been a new release since. I managed to give it try by compiling libdecor using [this aur package](https://aur.archlinux.org/packages/libdecor-git), and then by running following command, I got the GTK 3 window style:

```shell
WAYLAND_DISPLAY=wayland-1 GTK_THEME=adwaita wezterm
```

![Screenshot from 2023-06-19 22-59-27](https://github.com/wez/wezterm/assets/849640/c2909e7b-935f-495a-b991-f613b303542f)

**Comment by wez** [comment]
@arashm I think you're using XWayland there. As far as I understand it, you will need to modify either wezterm or SCTK to teach them to link to libdecor and cooperatively enable its decoration feature. If you haven't done the work for that, I don't see how it would change any behavior in wezterm.

**Comment by wez** [comment]
You can prove/disprove this by opening the debug overlay (CTRL-SHIFT-L) and looking at the `Window Environment` line. It will tell you whether it is using X11 or Wayland.

**Comment by arashm** [comment]
@wez I see. I think you're right. Sorry for that.

**Comment by martijk** [comment]
I know that this is not for everyone and I hope it's not regarded as useless noise, but the [Rounded Window Corners](https://github.com/yilozt/rounded-window-corners) GNOME extension adds a shadow to wezterm. If the lack of a shadow is a dealbreaker, this extension can be used to bridge the gap until there is a proper fix.
